File: Programmers/문자열 압축.py
def solution(s):
    answer = len(s)
    for i in range(1, len(s)//2+1):
        count = 1
        result = ''
        temp = s[:i]
        for j in range(i, len(s), i):
            if temp == s[j:j+i]:
                count+=1
            else:
                if count != 1:
                    result = result + str(count) + temp
                    count = 1
                else:
                    result = result + temp
                temp = s[j:j+i]
        if count != 1:
            result = result + str(count)+temp
        else:
            result = result + temp
        answer = min(answer, len(result))

    return answer

File: Programmers-JAVA/문자열 압축.java
class Solution {
    public int solution(String s) {
        int answer = Integer.MAX_VALUE;
        if(s.length() == 1)
            return 1;
        String temp = "";
        for(int i=1;i<s.length()/2+1;i++) {
            int count = 1;
            int slen = s.length();
            for(int j = i;j<=s.length();j+=i) {
                if(temp.equals(s.substring(j-i, j))) {
                    count++;
                }
                else {
                    if(count>1) {
                        slen-=i*(count-1);
                        slen+=(int)(Math.log10(count)+1);
                        count=1;
                    }
                }  
                temp = s.substring(j-i, j);
            }
            if(count > 1){
                slen-=i*(count-1);
                slen+=(int)(Math.log10(count)+1);
            }
            answer = Math.min(answer, slen);
        }
        return answer;
    }
}
